src/components/UserSelector.tsx

```diff
@@ -1,16 +1,56 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useRef, useState } from 'react';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+
+interface Props {
+  users: User[];
+  userId: number | null;
+  setUserId: React.Dispatch<React.SetStateAction<number | null>>;
+  setSelectedpost: React.Dispatch<React.SetStateAction<Post | null>>;
```

Comment:
The prop `setSelectedpost` should be renamed to `setSelectedPost` to maintain consistency with the naming convention used for state variables and props.